src/rules/no-multiple-stores.ts

```diff
@@ -0,0 +1,56 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils'
+
+import {
+  storeParameterWithoutVisibility,
+  storeParameterWithVisibility,
+} from './utils'
+
+export const ruleName = 'no-multiple-stores'
+
+export const messageId = 'noMultipleStores'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(name => name)<Options, MessageIds>({
+  name: ruleName,
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Possible Errors',
+      description: 'There should only be one store injected',
+      extraDescription: ['There is only one global store'],
+      recommended: 'error',
+    },
+    schema: [],
+    messages: {
+      [messageId]: 'Store should be injected only once',
+    },
+  },
+  defaultOptions: [],
+  create: context => {
```

Comment:
I wonder if we could do this by using just one query, something like this.
What do you think?

```ts
  create: context => {
    const constructors: TSESTree.Identifier[] = []

    return {
      [`MethodDefinition[kind='constructor'] Identifier>TSTypeAnnotation>TSTypeReference[typeName.name="Store"]`](
        node: TSESTree.TSTypeReference,
      ) {
        constructors.push(node.parent.parent as TSESTree.Identifier)
      },
      'Program:exit'() {
        constructors
          .filter((_, i) => i > 0)
          .forEach(node => {
            context.report({
              node,
              messageId,
            })
          })
      },
    }
  },
```

Reply:
The issue I could see is that with the @typescript-eslint/parser, a constructor parameter with a visibility modifier is one level deeper than a parameter without. (There is an additional node TSParameterProperty with a visibility modifier)

For the following code:
![image](https://user-images.githubusercontent.com/49377434/82811353-db4da700-9e90-11ea-96f0-e957817ea038.png)
ASTExplorer gives:
![image](https://user-images.githubusercontent.com/49377434/82811325-c96c0400-9e90-11ea-8063-c1b9310fd6ab.png)

So I'm not sure we can do everything with 1 query because the parents won't be se same.

I spent a lot of time trying to get one query but didn't succeed because of that.
I'll have a look to confirm this later today

Reply:
Ok, after a second look, I understand what you did there ! It is indeed much cleaner
I think we should consider renaming constructors to injectedStores. What do you think?

I couldn't find any documentation about the Program:exit hook. Do you know any?

Reply:
Would this code work if there are multiple classes in the same file?

Reply:
Yea, that's true that it doesn't work with multiple classes.
I think for now I would keep it as simple as possible, we can always add more logic later to it.
But as you mentioned, I also think it there won't be multiple classes in the same file.

👍 for the rename.

Reply:
Ok, I'll switch back to your implementation then

Reply:
I removed the multi classes management.
I kept the error report when we exit from the constructor instead of waiting for the end of the program.
I wonder if it has better performance. What do you think?

Reply:
That's awesome, thanks!